### DISCUSSION TITLE
Data pipeline configuration (user journey and audit trail)

### DISCUSSION THREAD

**Discussion by luizaraujobliver** [juspay/hyperswitch#3803] [General]
Hey guys, how are you?

We are configuring HS locally and the use of analytics functions are extremely important for the company. We need to use the user journey function, system metrics, audit trail and report generation.

It seems that for this we need to configure an entire data pipeline (kafka, zookeeper, ckh etc) but there is no documentation for this. Could you help us with this complete configuration of the data pipeline so that at least the user journey and audit trail work?

**Comment by su-shivanshmathur** [comment]
Hey @luizaraujobliver 
The documentation for setup of Kafka, ClickHouse, Zookeeper is already available 

Link for the same : https://github.com/juspay/hyperswitch/tree/main/crates/analytics/docs/clickhouse

**Reply by luizaraujobliver** [reply]
@ShivanshMathurJuspay We followed this tutorial, however, we had some problems along the way with some of the scripts not working. But I will try again and return with details of the errors.

Regarding the audit trail, does it depend on this configuration to work? Because for me it returns a SQLX configuration error, is it a different configuration of this pipeline?

![image](https://github.com/juspay/hyperswitch/assets/123188775/519d48af-b37a-4873-a3b6-f9d1da6f03ca)

![image](https://github.com/juspay/hyperswitch/assets/123188775/63c70351-d2ca-41ec-a67f-537a8e288a51)

![image](https://github.com/juspay/hyperswitch/assets/123188775/60580c07-a345-449b-a8cb-2fbedf2a1841)

**Reply by su-shivanshmathur** [reply]
Hey @luizaraujobliver 
Yes
With SQLX as the source for analytics
Audit trail won't work
Probably you would want to run dashboard without the feature flag `audit_trail` as `false` 
It would perfectly with Kafka and Clickhouse 
Lets get that working for better experience

**Comment by luizaraujobliver** [comment]
@ShivanshMathurJuspay We don't want to disable the flag because we need to use the audit trail. However, some scripts are giving errors, is it possible to provide a little more detail in the configuration doc? This audit trail feature is extremely important for us

**Comment by su-shivanshmathur** [comment]
@luizaraujobliver Can we then work on setting up the clickhouse and kafka pipeline 
with that audit trail will work fine

**Reply by luizaraujobliver** [reply]
Yes, we can. What is the guidance? @ShivanshMathurJuspay

**Reply by su-shivanshmathur** [reply]
Can you please re share the output of the https://github.com/juspay/hyperswitch/tree/main/crates/analytics/docs/clickhouse

**Comment by su-shivanshmathur** [comment]
@luizaraujobliver any luck with setting up the pipeline
